components/add_user_to_channel_modal/add_user_to_channel_modal.jsx

```diff
@@ -0,0 +1,307 @@
+import SuggestionList from 'components/suggestion/suggestion_list.jsx';
+
+import {placeCaretAtEnd} from 'utils/utils.jsx';
+
+export default class AddUserToChannelModal extends React.Component {
+    static propTypes = {
+
+        /**
+         * Function that's called when modal is closed
+         */
+        onHide: PropTypes.func.isRequired,
+
+        /**
+         * The user that is being added to a channel
+         */
+        user: PropTypes.object.isRequired,
+
+        /**
+         * Object used to determine if the user
+         * is a member of a given channel
+         */
+        channelMembers: PropTypes.object.isRequired,
+
+        actions: PropTypes.shape({
+
+            /**
+             * Function to add the user to a channel
+             */
+            addChannelMember: PropTypes.func.isRequired,
+
+            /**
+             * Function to fetch the user's channel membership
+             */
+            getChannelMember: PropTypes.func.isRequired,
+        }).isRequired,
+    }
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+
+            /**
+             * Whether or not the modal is visible
+             */
+            show: true,
+
+            /**
+             * Whether or not a request to add the user is in progress
+             */
+            saving: false,
+
+            /**
+             * Whether or not a request to check for the user's channel membership
+             * is in progress
+             */
+            checkingForMembership: false,
+
+            /**
+             * The user input in the channel search box
+             */
+            text: '',
+
+            /**
+             * The id for the channel that is selected
+             */
+            selectedChannelId: null,
+
+            /**
+             * An error to display when the add request fails
+             */
+            inviteError: '',
+        };
+        this.suggestionProviders = [new SearchChannelWithPermissionsProvider()];
+        this.enableChannelProvider();
+    }
+
+    enableChannelProvider = () => {
+        this.suggestionProviders[0].disableDispatches = false;
+    }
+
+    focusTextbox = () => {
+        if (this.channelSearchBox == null) {
+            return;
+        }
+
+        const textbox = this.channelSearchBox.getTextbox();
+        if (document.activeElement !== textbox) {
+            textbox.focus();
+            placeCaretAtEnd(textbox);
+        }
+    }
+
+    onChange = (e) => {
+        this.setState({text: e.target.value, selectedChannelId: null});
+    }
+
+    onHide = () => {
+        this.setState({show: false});
+        this.props.onHide();
+    }
+
+    setSearchBoxRef = (input) => {
+        this.channelSearchBox = input;
+        this.focusTextbox();
+    }
+
+    handleInviteError = (error) => {
+        if (error) {
+            this.setState({inviteError: error.message, saving: false});
+            return;
+        }
+        this.onHide();
```

Comment:
I agree. I had the call in the `handleInviteError` function because that is where the error was being asserted.

I am going with this strategy instead:

```
this.props.actions.addChannelMember(channelId, user.id).then(({error}) => {
  if (error) {
    this.handleInviteError(error);
  } else {
    this.onHide();
  }
});
```

Reply:
Hi @mickmister, I agree with your strategy above.  Can you help push another with that change?  Thanks!